thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -430,10 +430,13 @@ void CDiskWriteSlaveActivityBase::close()
             outraw.clear();
         }
 
+        Owned<IFileIO> tmpFileIO;
         {
             CriticalBlock block(statsCs);
             mergeStats(fileStats, outputIO);
-            outputIO.clear();
+
+            // ensure it is released/destroyed after releasing crit, since the IFileIO might involve a final copy and take considerable time.
+            tmpFileIO.setown(outputIO.getClear());
         }
 
         if (!rfsQueryParallel && dlfn.isExternal() && !lastNode())
```

Comment:
@jakesmith Should this be done before or after the file is destroyed?  If after then there should be a tmpFileIO.clear()

Reply:
so this is only in the case where the file is external (file:://) and all nodes are extending the same temporary output (on node 1). The last node will do the rename or copy to the final destination.

Triggering (to the next node) asap is preferrable, but it won't actually make any difference whether before or after, since it's only the last node that will do anything on the destruction of tmpFileIO.

@ghalliday - I'll rebase to 7.4.x